src/loader/loader.js

```diff
@@ -35,17 +35,17 @@
          * check the loading status
          * @ignore
          */
-        function checkLoadStatus() {
+        function checkLoadStatus(onload) {
             if (loadCount === resourceCount) {
                 // wait 1/2s and execute callback (cheap workaround to ensure everything is loaded)
-                if (api.onload) {
+                if (onload || api.onload) {
                     // make sure we clear the timer
                     clearTimeout(timerId);
                     // trigger the onload callback
-
-                    var onload = api.onload;
+                    // we call either the supplied callback (which takes precedence) or the global one
+                    var callback = onload || api.onload;
```

Comment:
Excellent observation! The function is private (undocumented), so none of the fallback logic is even necessary.